pkg/tfbridge/info.go

```diff
@@ -148,6 +148,9 @@ type SchemaInfo struct {
 
 	// to remove empty object array elements
 	SuppressEmptyMapElements *bool
+
+	// this will make the parameter as computed and not allow the user to set it
+	MarkAsComputed *bool
```

Comment:
Computed doesn’t normally mean “output only”. It means “also output”. Right?

Per the issue this is trying to address - the property in question is computed + optional. It’s the optional piece that causes the problem I believe.

Reply:
We can’t remove the optional - TF scheme requires something to be optional + computer, required or computed and some combination

This new override will drop anything else on that specific field and make that parameter computed only - it effectively forgets that the optional is there (or rather, that’s the intent)

This gives us control where these optional + computed happens. There are a lot of data sources that do this

Reply:
The way we interpret these values is like so:
1. `Computed` only: this property only exists in the resource's state. It is not an input property.
2. `Optional` or `Required`: this property is an input, and its input value (if present) is also copied verbatim to the state.
3. `Optional` or `Required` _plus_ `Computed`: this property might be an input or might only be present in the state. If the property is an input, it is possible that the value in the state may differ (though in general this should not be the case w/o `CustomizeDiff`, as differing values will cause spurious changes in `Diff`).

What we want here is the ability to treat a property as if it was marked `Computed` only.

Reply:
👍 exactly that!